### DISCUSSION TITLE
MiniDexed on a Le Potato?

### DISCUSSION THREAD

**Discussion by agent211** [probonopd/MiniDexed#410] [Q&A]
I been trying to run a MiniDexed on a Libre Le Potato.  Similar to a RPi 3, but uses a Amlogic S905X instead of the Broadcom BCM2837 (both still use the Cortex A53).  Bare metal OS's are not really my forte and I haven't had any luck.  I've been able to load Linux distributions on the Potato, so I'm confident the hardware is OK, but that's as far as I've gotten.  Any tips, tricks, pointers, ideas would be greatly appreciated.  

Thanks.

**Comment by BobanSpasic** [comment]
I guess that the right address for your question would be in **circle** repository:
https://github.com/rsta2/circle

as I can see from the discussions there, there is not much hope that circle would be ported to other platforms without volunteers with some knowledge how to do it:
https://github.com/rsta2/circle/discussions/327

**Reply by agent211** [reply]
Thanks for that lead.  You saved me a lot of time.  Looks like it's not possible, short of trying to write and compile my own OS from scratch, while successfully porting over the MiniDexed stuff.  I'd actually have a better chance of building a TX816 from scratch.  😆

**Comment by probonopd** [comment]
Possibly one aproach would be to port MiniDexed to the “naked“ Linux kernel (not a Linux distribution).

**Comment by dcoredump** [comment]
The base engine "Dexed" is also working as a LV2- Plugin (see https://github.com/dcoredump/dexed.lv2).

Dexed.lv2 is used by Zynthian (https://zynthian.org). With Zynthian you have already a ready to use (and open) Linux based synth and you can easily build your own TX by creating 8 chains of Dexed.lv2 and add as many FX as your CPU time (and RAM) allows.

Of course, this is RPi based...

**Reply by probonopd** [reply]
I tried Zynthian and it is a full Linux distribution, gigabytes full of software, and based on Jack (which is a safe guarantee that everything will be complicated and nothing will work right out of the box without lengthy configuration). I ran into issue after issue and everything was dog slow. What I am thinking about is the combination of just the Linux kernel plus one binary containing MiniDexed; nothing else. No init, no Jack, etc. Maybe others had better experiences with Zynthian, but if you are after an "embedded device"/"real instrument" feeling, then I don't think it's a good match.

**Reply by dcoredump** [reply]
Strange, when was the last time you tried Zynthian? I have several Zynthians and even use one on stage. The actual software is pretty good and using jack/pipewire in the background works pretty well too.

Do you really think it's worth to put the audio buffers from dexed directly into the Linux kernel? In the end it is almost easier to use [MicroDexed-](https://codeberg.org/dcoredump/MicroDexed.git) or [MicroDexed-touch](https://codeberg.org/positionhigh/MicroDexed-touch.git) based on a [Teensy](https://www.pjrc.com/).
